server/hwapi/data_models/models.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Canonical Ltd.
+# All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Written by:
+#        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+
+
+from datetime import date, datetime
+
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
+from sqlalchemy.orm import relationship
+
+
+class Base(DeclarativeBase):
+    """Base model for all the models"""
+
+    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+
+
+class Vendor(Base):
+    __tablename__ = "vendors"
+    name: Mapped[str] = mapped_column(unique=True, nullable=False)
+
+
+class Platform(Base):
+    __tablename__ = "platforms"
+    name: Mapped[str] = mapped_column(unique=True, nullable=False)
+    vendor_id: Mapped[int] = mapped_column(ForeignKey("vendors.id"))
+    vendor: Mapped[Vendor] = relationship(foreign_keys=[vendor_id])
```

Comment:
Not sure if this form of relationship configuration works. Perhaps it would be safer to follow the SQLAlchemy [docs](https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html)

Reply:
I've used the information from [this doc](https://docs.sqlalchemy.org/en/20/orm/join_conditions.html), but since our relations here are more obvious, we can use your suggestion here